torch/optim/adadelta.py

```diff
@@ -268,6 +268,10 @@ def _single_tensor_adadelta(
             for p, step in zip(params, state_steps)
         ), f"If capturable=True, params and state_steps must be on supported devices: {capturable_supported_devices}."
 
+    if isinstance(lr, Tensor):
```

Comment:
let's replace all of these with lr = _to_scalar(lr)

Reply:
I had used `_to_scalar()` in `_single_tensor_adadelta`, but replaced it with the current code because it caused a `jit.script` issue. Do you know how to support for `jit.script`?

https://github.com/pytorch/pytorch/actions/runs/13330919253/job/37239329378

<img width="732" alt="_to_scalar_in_adadelta" src="https://github.com/user-attachments/assets/065ff7c9-161c-43b6-8e5d-fc4ea67c0bfb" />

Reply:
ahhh torchscript is so annoying. let's just do something like this:

```
if not torch.jit.is_scripting():
	lr = _to_scalar(lr)
```

Reply:
I refactored them using `_to_scalar`: https://github.com/pytorch/pytorch/pull/145674/commits/ee7acf5160995716f73ca5f18f6220d6cf750bf9